<!-- -->

<launch>
	<!-- REQUIED by mk_interface -->
	<arg name="qcID" value="1" />
	<arg name="firmware" value="5" />
	<arg name="serial_device1" value ="ttyUSB0" /> <!-- interceptty -->

	<arg name="serial_device2" value ="usb-FTDI_FT232R_USB_UART_A900fQqb-if00-port0" />

	<!-- REQUIED by telekyb_core -->
	<arg name="rotortype" value="Quadcopter" />
	<arg name="takeOffHeight" value="-1" />
	<arg name="landHeight" value="-0.25" />
	<arg name="initialMass" value="0.7" />
	<arg name="emergencyThrust" value="60" />

	<!-- Vicon -->
	<!-- <node name="ViconEmulator_$(arg qcID)" pkg="telekyb_vicon_emulator" type="vicon_emulator" args="$(arg qcID)"/>-->


	<!-- mk interface  -->
	<node name="mk_omegacontrolinterface_$(arg qcID)" pkg="tk_mkomegacontrolinterface" type="mkomegacontrol">
		<param name="MKOmegaControlInterface/tUavId" value="$(arg qcID)" />
		<param name="MKOmegaControlInterface/tUavFirmware" value="$(arg firmware)"/>
		<param name="MKOmegaControlROSInterface/tCommandsTopic" value="Commands"/>
		<param name="MKOmegaControlROSInterface/RCCommandsTopic" value="RCCommands"/>
		<param name="MKOmegaControlROSInterface/operatorRequestTopic" value="operator"/>
		<param name="MKOmegaControlROSInterface/tEmergencyLandService" value="EmergancyLand" />
		<param name="tEmergencyThrust" value="60" />
		<param name="MKOmegaControlInterface/tCommandType" value="omega"/>
		<param name="MKOmegaControlInterface/tRepublishBlCommands" value="true"/>
		<!-- Serial -->
		<param name="MKOmegaControlInterface/tSerialDeviceDirectory" value="/dev/" /> <!-- /tmp/ -->
		<param name="MKOmegaControlInterface/tSerialDeviceNameRegex" value="$(arg serial_device1)" />
		<param name="MKOmegaControlInterfaceConnection/tBaudRate" value="BAUD115200" />
		<!--<param name="MKOmegaControlInterfaceConnection/tBaudRate" value="BAUD500000" />-->
	</node>


</launch>
